### ISSUE TITLE
tests for correctness

### ISSUE THREAD

**Issue by skchoi56** [rbraju/cribl-tests#2]
can some be added to verify splitter behaves as desired? thanks!

**Comment by rbraju** [comment]
I couldn't find any info in splitter about the data processing. I see below messages in docker logs which mentions it connected to targets successfully. But couldn't find a way to determine if it completed sending the data to the targets. I can add tests looking for logs that confirms the connectivity.

```
My hostname is: c61fd6d61140
working as splitter
targets [ { host: 'target_1', port: 9997 }, { host: 'target_2', port: 9997 } ]
App listening on port 9997

client connected
processing  { host: 'target_1', port: 9997 }
processing  { host: 'target_2', port: 9997 }
Connected to  { host: 'target_2', port: 9997 }
Connected to  { host: 'target_1', port: 9997 }
```

**Comment by rbraju** [comment]
Added few assertions to validate the connectivity between agent-->splitter and splitter-->target.

**Comment by skchoi56** [comment]
Thanks @rbraju. By `correctness` I specifically meant data received by the targets is correct -- anything strange about the data received at target_1 and target_2?

re: `couldn't find a way to determine if it completed sending the data to the targets`
^ anything in the "behavior" of the agent node?

**Comment by rbraju** [comment]
@skchoi56 
Thanks for the response.
I added a test to verify if the data received by target is accurate. By comparing it with the source. I couldn't do that with 1M events and hence downsized the input to 50k. 

I noticed that are some unprocessed events that happens intermittently (Like few events from the source are not found in any of the targets). I couldn't reproduce that consistently. I was thinking that may be an issue with Splitter.

May be can have another test to validate that agent --> splitter data transfer is accurate before validating the accuracy in target nodes.

**Comment by skchoi56** [comment]
Possible to explain: "I couldn't do that with 1M events and hence downsized the input to 50k" ? You can add more input files for testing, but ideally would validate the 1M original input file.

No need to add tests on splitter if you add tests on the targets.

**Comment by rbraju** [comment]
Since the file content is huge and randomly split, the code I wrote took too long to compare the content from target nodes with the source. Hence I used a file with smaller size to see if all the events from the source is processed.

**Comment by skchoi56** [comment]
Please keep the original 1M event file as the "base" test case. (Any extra files are bonus)

To compare content from target with source: are there patterns to check, given you can see the input?

**Comment by rbraju** [comment]
Updated the agent and tests to use 1M event file as a base. Sorry that I couldn't think of a pattern on top of my mind, but I can try. I was comparing all the entires from the source with the targets. Hence commented out the 'data accuracy' test as it is taking long time for line by line comparison.

**Comment by rbraju** [comment]
Yes :)
My assumption is that the input file can contain any raw data (like logs, events or traces etc, and may not follow a common pattern throughout the file). Hence wanted to test in a more generic way like string comparison. ​Also, I noticed that some of the lines were not properly carried to the target nodes even though the # of lines (from all target nodes combined) matches with the source data size.
​
For example,
```
[rajesh:/opt/cribl/cribl-tests]: grep "195882" test-artifacts/large_1M_events.log 
This is event number 195882

[rajesh:/opt/cribl/cribl-tests]: grep "195882" test-artifacts/target_1_events.log 
This is ember 195882
```

For this particular example, we can find the common substring from the input list and look for it in the target nodes. But still need to validate the rest of the string along with it.  I've added a test to validate if the common pattern is available in target nodes and list out the odd ones.
